### PR TITLE
♻️ Maintenance: Fixes mypy in api-server

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Following procedure in #6115, this PR fixes errors reported by `mypy` in `api-server` service

## Related issue/s

- Part of #6115

## How to test

```
cd services/api-server
make install-dev
make mypy
```

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/api-server/src/simcore_service_api_server/api/routes/files.py [hunk 1]
```diff
@@ -353,7 +354,7 @@ async def abort_multipart_upload(
     file: File = File(
         id=file_id,
         filename=client_file.filename,
-        sha256_checksum=client_file.sha256_checksum,
+        checksum=client_file.sha256_checksum,
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/api/routes/files.py:357]: isn't this a good one? using the wrong arguments?

File: services/api-server/src/simcore_service_api_server/api/routes/meta.py [hunk 2]
```diff
@@ -16,8 +16,8 @@ async def get_service_metadata(
 ):
     return Meta(
         name=__name__.split(".")[0],
-        version=API_VERSION,
-        released={API_VTAG: API_VERSION},
+        version=API_VERSION,  # type: ignore[arg-type]
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/api-server/src/simcore_service_api_server/api/routes/meta.py:19]: check the director-v2 for an example -> no need to ignore that one

File: services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py [hunk 3]
```diff
@@ -389,7 +392,10 @@ async def get_job_wallet(
     _logger.debug("Getting wallet for job '%s'", job_name)
 
     if project_wallet := await webserver_api.get_project_wallet(project_id=job_id):
-        return await webserver_api.get_wallet(wallet_id=project_wallet.wallet_id)
+        return cast(
+            WalletGetWithAvailableCredits,
+            await webserver_api.get_wallet(wallet_id=project_wallet.wallet_id),
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py:397]: Here I believe that the decorator ```_exception_mapper``` around webserver_api.get_wallet is obscuring the types.
In order to properly fix this you can do something similar as in https://github.com/itisfoundation/osparc-simcore/blob/e238d3e705ede10e497a3614890e3eb3d48388e7/packages/aws-library/src/aws_library/ssm/_error_handler.py that uses typing.ParamSpec and typing.TypeVar.
This makes both Pylance and mypy understand these decorated function. pretty cool stuff. almost feels like C++ templating

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py [hunk 4]
```diff
@@ -116,7 +116,8 @@ async def create_study_job(
     )
 
     await webserver_api.patch_project(
-        project_id=job.id, patch_params=ProjectPatch(name=job.name)
+        project_id=job.id,
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py:119]: I guess here might be the same problem with that decorator of yours

File: services/api-server/src/simcore_service_api_server/models/domain/projects.py [hunk 5]
```diff
@@ -1,8 +1,10 @@
-# mypy: disable-error-code=truthy-function
-from models_library.projects import AccessRights, Node, StudyUI
-from models_library.projects_nodes import InputTypes, OutputTypes
+from models_library.projects_access import AccessRights
+from models_library.projects_nodes import InputTypes, Node, OutputTypes
 from models_library.projects_nodes_io import SimCoreFileLink
 
+# mypy: disable-error-code=truthy-function
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/models/domain/projects.py:5]: Question: why you do not put this on top?

File: services/api-server/src/simcore_service_api_server/models/schemas/files.py [hunk 6]
```diff
@@ -146,7 +146,9 @@ def create_id(cls, *keys) -> UUID:
     @property
     def storage_file_id(self) -> StorageFileID:
         """Get the StorageFileId associated with this file"""
-        return parse_obj_as(StorageFileID, f"api/{self.id}/{self.filename}")
+        return parse_obj_as(
+            StorageFileID, f"api/{self.id}/{self.filename}"  # type: ignore[arg-type]
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/models/schemas/files.py:150]: let's hope for pydantic v2

File: services/api-server/src/simcore_service_api_server/services/solver_job_models_converters.py [hunk 7]
```diff
@@ -60,18 +60,19 @@ def create_node_inputs_from_job_inputs(
 
     node_inputs: dict[InputID, InputTypes] = {}
     for name, value in inputs.values.items():
-        assert parse_obj_as(ArgumentTypes, value) == value  # type: ignore  # nosec
+        assert parse_obj_as(ArgumentTypes, value) == value  # nosec # type: ignore
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/services/solver_job_models_converters.py:63]: ignore is missing specs

File: services/api-server/src/simcore_service_api_server/services/solver_job_models_converters.py [hunk 8]
```diff
@@ -135,8 +138,8 @@ def create_new_project_for_job(
     )
 
     solver_service = Node(
-        key=solver.id,
-        version=solver.version,
+        key=solver.id,  # type: ignore[arg-type]
+        version=solver.version,  # type: ignore[arg-type]
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/services/solver_job_models_converters.py:142]: a bit suspicous this no?

File: services/api-server/src/simcore_service_api_server/api/routes/files.py [hunk 9]
```diff
@@ -382,7 +383,7 @@ async def complete_multipart_upload(
...
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/routes/files.py:386]: see message above. These were covered by tests as well

File: services/api-server/src/simcore_service_api_server/api/routes/meta.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/routes/meta.py:19]: @sanderegg this is one of the issues I wanted to discuss with you offline. Here I leave some thoughts:

- A constructor or constraint type does not validatee
```python
coverted_to_type_but_not_validated = VersionStr(API_VERSION)
converted_to_type_and_validated = parse_obj_as(VersionStr, API_VERSION)
```
- `Meta` will do the parsing and conversion. Why then to use any of the above?

File: services/api-server/src/simcore_service_api_server/api/routes/meta.py [hunk 10]
```diff
@@ -16,8 +16,8 @@ async def get_service_metadata(
...
+        released={API_VTAG: API_VERSION},  # type: ignore[dict-item]
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/routes/meta.py:20]: I have the same questions as above. We have to look into this :-)

File: services/api-server/src/simcore_service_api_server/models/api_resources.py [hunk 11]
```diff
@@ -41,9 +41,11 @@ class Config:
 
 
 def parse_last_resource_id(resource_name: RelativeResourceName) -> str:
-    match = RelativeResourceName.regex.match(resource_name)
-    last_quoted_part = match.group(1)
-    return urllib.parse.unquote_plus(last_quoted_part)
+    if match := RelativeResourceName.regex.match(resource_name):
```
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/models/api_resources.py:44]: Thisone is a good catch

File: services/api-server/src/simcore_service_api_server/models/domain/projects.py (hunk 5 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/models/domain/projects.py:5]: class issue with comments between imports ... when they are automatically rearranged it gets in the middle

File: services/api-server/src/simcore_service_api_server/models/schemas/files.py (hunk 6 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/models/schemas/files.py:150]: Amen

File: services/api-server/src/simcore_service_api_server/services/solver_job_models_converters.py (hunk 7 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/services/solver_job_models_converters.py:63]: Do you really think it is so relevant when it comes to asserts??

File: services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py (hunk 3 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py:397]: Thx. Really nice input!

**Review by matusdrobuliak66 — APPROVED** [review]
thanks

**Review by giancarloromeo — APPROVED** [review]

**Review by GitHK — APPROVED** [review]
👍 thanks

**Pull request merged**